### ISSUE TITLE
[IOS] [EAS] Error loading page : Domain: NSURLErrorDomain - Error Code - 1003

### ISSUE THREAD

**Issue by RainLdc** [expo/eas-cli#1009]
### Summary

When i use 
eas build --platform ios, then transporter to upload on app store, i have the following error on opening app on iphone.

I use extra in app.

My app.config.js

```js
export default {
  [.....]
  updates: {
    fallbackToCacheTimeout: 0,
  },
  assetBundlePatterns: [
    '**/*',
  ],
  extra: {
    NODE_ENV: process.env.NODE_ENV,
    url: config.get('frello.url.url'),
  },
};
```

```js
import Constants from 'expo-constants';

export const url = Constants.manifest.extra.url;
```

### Managed or bare?

Managed

### Environment

```
  expo-env-info 1.0.2 environment info:
    System:
      OS: macOS 12.0.1
      Shell: 5.8 - /bin/zsh
    Binaries:
      Node: 16.5.0 - ~/.nvm/versions/node/v16.5.0/bin/node
      npm: 8.3.0 - ~/.nvm/versions/node/v16.5.0/bin/npm
      Watchman: 2021.06.07.00 - /usr/local/bin/watchman
    SDKs:
      iOS SDK:
        Platforms: DriverKit 21.2, iOS 15.2, macOS 12.1, tvOS 15.2, watchOS 8.3
    IDEs:
      Xcode: 13.2.1/13C100 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^44.0.0 => 44.0.6
      react: 17.0.1 => 17.0.1
      react-dom: 17.0.1 => 17.0.1
      react-native: 0.64.3 => 0.64.3
    npmGlobalPackages:
      eas-cli: 0.47.0
      expo-cli: 5.3.0
    Expo Workflow: managed
```

### Error output

Error loading page : Domain: NSURLErrorDomain - Error Code - 1003
Description : A server with the speicified hotsname cold not be found.

### Reproducible demo or steps to reproduce from a blank project

![IMG_4240](https://user-images.githubusercontent.com/25221245/157668668-aca7aa77-e172-44bf-b98f-073684f8ba3b.PNG)

**Comment by RainLdc** [comment]
Docs : 

- https://docs.expo.dev/build-reference/variables/
- https://docs.expo.dev/build/eas-json/

**Comment by wkozyra95** [comment]
Please provide link to the build page

>     NODE_ENV: process.env.NODE_ENV,

I don't know why are you doing this, but most likely this is a mistake

> config.get('frello.url.url'),

Is this the URL that is missing? what is the config object ? try adding conole log in app.config.js and check eas build logs

**Comment by RainLdc** [comment]
I found the solution.

In eas.json, the doc add APP_ENV in env. But for config.js i need NODE_ENV.
So i just need to add NODE_ENV in eas.json.
Thank you.

```js
{
    "cli": {
        "version": ">= 0.47.0"
    },
    "build": {
        "production": {
            "env": {
                "NODE_ENV": "production"
            }
        }
    },
}
```

**Comment by wkozyra95** [comment]
You should not use NODE_ENV variable to configure the environment, that variable has special meaning for node, npm, packager and can affect the build, so unless you explicitly want to trigger that behavior use some custom value like APP_ENV

**Comment by RainLdc** [comment]
> You should not use NODE_ENV variable to configure the environment, that variable has special meaning for node, npm, packager and can affect the build, so unless you explicitly want to trigger that behavior use some custom value like APP_ENV

But if I use config.js, I need to use NODE_ENV to use the right config during the build no ?

**Comment by wkozyra95** [comment]
no, you can use any variable you want, but using NODE_ENV is a bad idea because that variable has a special meaning for different node and node-based tools
